sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/OkHttp3ClientInterceptor.java

```diff
@@ -93,4 +107,36 @@ private Optional<Request> getRequest(ExecuteContext context) {
         }
         return Optional.empty();
     }
+
+    private Map<String, String> getHeaders(Headers headers) {
+        Map<String, String> headerMap = new HashMap<>();
+        for (String name : headers.names()) {
+            headerMap.putIfAbsent(name, headers.get(name));
+        }
+        return headerMap;
+    }
+
+    private Request rebuildRequest(Request request, ServiceInstance serviceInstance) {
+        return request.newBuilder()
+                .url(HttpUrl
+                        .parse(BaseHttpRouterUtils
+                                .rebuildUrlByXdsServiceInstance(request.url().uri(), serviceInstance)))
+                .build();
+    }
+
+    private void handleXdsRouterAndUpdateHttpRequest(Object obj, Request request) {
+        Headers headers = request.headers();
+        URI uri = request.url().uri();
+        String host = uri.getHost();
+
+        // use xds route to find a service instance, and modify url by it
+        if (routerConfig.isEnabledXdsRoute() && BaseHttpRouterUtils.isXdsRouteRequired(host)) {
```

Comment:
I think ‘routerConfig.isEnabledXdsRoute()’ can be evaluated at first so you do not have to create previous objects

Reply:
done